### ISSUE TITLE
Creating A Tenant Creates A Dummy Resource

### ISSUE THREAD

**Issue by never00rei** [auth0/terraform-provider-auth0#686]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Tenant resource creates a dummy resource which can be confusing unless one has imported an already existing tenant as a resource.

### Expectation

When creating a new tenant block, terraform should throw an error as creating a tenant is not currently available in the Auth0 management API.

### Reproduction

1. Create new tenant block with a new name.
2. Run terraform.
3. Terraform state represents a new tenant.

### Auth0 Terraform Provider version

0.47.0

### Terraform version

v1.5.2

**Comment by sergiught** [comment]
Hey @never00rei 👋🏻 

Currently, the `auth0_tenant` resource does not support the creation of new tenants. If this resource is not imported, it will automatically assign a unique ID to the resource and perform an update. This approach is intended to enhance the developer experience by eliminating the need for importing, as the `auth0_tenant` resource is designed to be a singleton.

It is something we mention in the notes within the docs as well: https://registry.terraform.io/providers/auth0/auth0/latest/docs/resources/tenant.

Would you mind clarifying further the issue you're experiencing?

**Comment by never00rei** [comment]
Ok, I have some queries surrounding your reply to this issue:

1. Can you elaborate on your use of `singleton` please, are you expecting a single instantiation of `auth0_tenant` per repository, module or Terraform workspace for instance?
2. Are you fully expecting the Auth0 Management API to throw an Error if the referenced tenant does not exist?

Something that also worries me is that you mention that: 
```If this resource is not imported, it will automatically assign a unique ID to the resource and perform an update. This approach is intended to enhance the developer experience by eliminating the need for importing```
Importing a resource into the Terraform state is a much _safer_ operation over just carte blanche updating something without understanding _what you're potentially going to change_. To me this approach seems like the complete opposite to what the intended workflow was designed to be in Terraform with regards to managing the current and intended future state of a resource.

**Comment by sergiught** [comment]
Hey @never00rei 👋🏻 

Happy to clarify further and see if we can improve things together as an outcome of the discussion.


> Can you elaborate on your use of singleton please, are you expecting a single instantiation of auth0_tenant per repository, module or Terraform workspace for instance?

Within the Auth0 Infrastructure there are certain resources that are not creatable, only configurable. The Tenant is one such resource, other examples include Attack Protection, Branding, Guardian etc. Creatable resources are things like Clients, Connections, Roles, Users etc. 

You can only ever configure one tenant at a time through this provider. To configure another one you'd need to instantiate another instance of the provider and pass a different domain and client credentials, for example:

``` terraform
provider "auth0" {
  domain        = "<domain>"
  client_id     = "<client-id>"
  client_secret = "<client-secret>"
}

resource "auth0_tenant" "my_tenant" {
 # ... config
}
```

To work on multiple tenants you'd need to manage multiple workspaces each with its own state file. 


> Are you fully expecting the Auth0 Management API to throw an Error if the referenced tenant does not exist?
Yes, the Auth0 Management API will throw an error in case the tenant does not exist. The tenant on which you're applying the configuration is defined within the provider block. 

> Importing a resource into the Terraform state is a much safer operation over just carte blanche updating something without understanding what you're potentially going to change. To me this approach seems like the complete opposite to what the intended workflow was designed to be in Terraform with regards to managing the current and intended future state of a resource.

Completely agree, importing is a much safer operation as after importing you're presented with an accurate diff between local config and remote. However in the specific case of the auth0_tenant resource, only the options that are present in the config will get updated, all the other ones will not get reset to empty or removed. So to be more concrete if you have the following config:

```terraform
resource "auth0_tenant" "my_tenant" {
  friendly_name = "Travel0"
}
```

And you perform a `terraform apply` without having imported this resource, only the `friendly_name` attribute will get updated, nothing else will get changed as the other attributes are optional. You need to opt in to whatever you want to manage on these single instance resources.

**Comment by sergiught** [comment]
Closing this down as the current behavior is intended. Happy to resume the discussion if any other feedback arises.

Also we just shipped the [v1.0.0-beta.0](https://registry.terraform.io/providers/auth0/auth0/latest/docs) release! 🙌🏻 Please give it a try and let us know if you encounter any issues.

```
terraform {
  required_providers {
    auth0 = {
      source  = "auth0/auth0"
      version = "1.0.0-beta.0"
    }
  }
}
```

### 📚 Resources

[Quickstart](https://registry.terraform.io/providers/auth0/auth0/latest/docs/guides/quickstart) | [Documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) | [Migration Guide](https://github.com/auth0/terraform-provider-auth0/blob/v1/MIGRATION_GUIDE.md) | [Changelog](https://github.com/auth0/terraform-provider-auth0/releases/tag/v1.0.0-beta.0)
